main/lsp/LSPIndexer.h

```diff
@@ -25,34 +25,39 @@ class LSPIndexer final {
     /** Global state that we keep up-to-date with file edits. We do _not_ typecheck using this global state! We clone
      * this global state every time we need to perform a slow path typechecking operation. */
     std::unique_ptr<core::GlobalState> initialGS;
-    /** Contains file hashes for the files stored in `initialGS`. Used to determine if an edit can be typechecked
-     * incrementally. */
-    std::vector<core::FileHash> globalStateHashes;
     /** Contains a copy of the last edit committed on the slow path. Used in slow path cancelation logic. */
     LSPFileUpdates pendingTypecheckUpdates;
     /** Contains a clone of the latency timer for the pending typecheck operation. Is used to ensure that we correctly
      * track the latency of canceled & rescheduled typechecking operations. */
     std::unique_ptr<Timer> pendingTypecheckLatencyTimer;
-    /** Contains globalStateHashes evicted with `pendingTypecheckUpdates`. Used in slow path cancelation logic. */
-    UnorderedMap<int, core::FileHash> pendingTypecheckEvictedStateHashes;
+    /** Contains files evicted by `pendingTypecheckUpdates`. Used to make fast path decisions in the immediate past. */
+    UnorderedMap<int, std::shared_ptr<core::File>> evictedFiles;
     std::unique_ptr<KeyValueStore> kvstore; // always null for now.
     /** A WorkerPool with 0 workers. */
     std::unique_ptr<WorkerPool> emptyWorkers;
 
-    std::vector<core::FileHash> computeFileHashes(const std::vector<std::shared_ptr<core::File>> &files,
-                                                  WorkerPool &workers) const;
+    void computeFileHashes(const std::vector<std::shared_ptr<core::File>> &files, WorkerPool &workers) const;
+
+    /** Determines if the given edit can take the fast path relative to the most recently committed edit. If
+     * `containsPendingTypecheckUpdates` is `true`, it will make the determination in the immediate past using
```

Comment:
For my own understanding:  all of this comment makes sense to me except for the phrase "in the immediate past"...given that the determination is done "relative to the most recently committed edit", is "the immediate past" redundant or is it communicating new information?  (Is there some other way to calculate relative to the most recently committed edit that isn't the immediate past?)

Reply:
It's not redundant! It's new information.

"The immediate past" is "prior to the currently running slow path". I can clarify the comment.